packages/client/src/runtime/core/types/GetResult.ts

```diff
@@ -48,21 +48,19 @@ export type FluentOperation =
 type Count<O> = { [K in keyof O]: Count<number> } & {}
 
 // prettier-ignore
-export type GetFindResult<P extends Payload, A> =
-  {} extends A ? DefaultSelection<P> :
-  A extends 
-  | { select: infer S } & Record<string, unknown>
-  | { include: infer S } & Record<string, unknown>
-  ? S extends undefined ? DefaultSelection<P> :
-    {
-      [K in keyof S as S[K] extends false | undefined | null ? never : K]:
-        S[K] extends object
+export type GetFindResult<P extends Payload, _A, A = { 0: _A, 1: { select: undefined } }[Equals<_A, any>]> =
+  A extends
+  | { select: infer S extends object } & Record<string, unknown>
+  | { include: infer I extends object } & Record<string, unknown>
+  ? {
+      [K in keyof S | keyof I as (S & I)[K] extends false | undefined | null ? never : K]:
```

Comment:
Would this work / be less ambiguous?

```suggestion
      [K in (keyof S | keyof I) as (S & I)[K] extends false | undefined | null ? never : K]:
```